ci/src/mainnet_revisions/mainnet_revisions.py

```diff
@@ -20,6 +21,47 @@ class Command(Enum):
     CANISTERS = 2
 
 
+def base_download_url(git_commit_id, variant, update, test, dev=True):
+    """Construct the base download URL for ICOS images."""
+    component = ("update-img" if update else "disk-img") + ("-dev" if dev else "")
+    test_suffix = "-test" if test else ""
+    return f"https://download.dfinity.systems/ic/{git_commit_id}/{variant}/{component}{test_suffix}/"
+
+
+# NOTE: We must download and hash the update dev image ourselves because public proposals only include hashes for the production images
+def download_and_hash_image(url, logger):
+    """Download an image from the given URL and return its SHA256 hash."""
+    logger.info(f"Downloading image from: {url}")
+
+    try:
+        with urllib.request.urlopen(url, timeout=300) as response:
```

Comment:
Shouldn't we check for the response status using something like:
```suggestion
        with urllib.request.urlopen(url, timeout=300) as response:
            status = response.status 
            if status!= 200:
                raise Exception(f"Unexpected response {status}!")
```
I also wonder what happens when we get a 300 redirect.